applications/ParticleMechanicsApplication/custom_constitutive/hyperelastic_3D_law.cpp

```diff
@@ -153,10 +153,19 @@ void HyperElastic3DLaw::InitializeMaterial( const Properties& rMaterialPropertie
         const GeometryType& rElementGeometry,
         const Vector& rShapeFunctionsValues )
 {
-  mDeterminantF0                = 1;
-  mInverseDeformationGradientF0 = IdentityMatrix(3);
-  mStrainEnergy                 = 0;
+    bool restarted = false;
+    
+    if (rShapeFunctionsValues.size() > 0){
+        if (rShapeFunctionsValues(0) != -888 ){
```

Comment:
this is quite a hack :thinking: 
@AlejandroCornejo how do you handle this in the CLawsApp?
This function seems to be one of the few (if not the only one) that does not have access to the `ProcessInfo`

Reply:
Yes 😅 i've been trying to figure out a way to solve this differently, for example by adding a "IS_RESTARTED" value to the Material Properties but I was running into issues when running it with multiple threads in OMP. So any other suggestions are welcome, I just used this as a temporary solution until I could come up with something different.

Reply:
I did not look at it in detail, but I know that this would not be the first function in Elements/Conditions/Constitutive-Laws that is missing the `ProcessInfo

Lets wait for the response of @AlejandroCornejo
The proper solution would probably be to update the interface of this function. This involves the core and hence takes some time, but we have a workflow/knowledge how to do it backward compatible (e.g. see #2993)

Reply:
> Material Properties but I was running into issues when running it with multiple threads in OMP

yes this is not allowed, and will lead to race conditions as you experienced :)